src/rendering/helper.cpp

```diff
@@ -353,5 +355,37 @@ void renderBox(const glm::vec2& position, const glm::vec2& size, const glm::vec4
     shdr.program->deactivate();
 }
 
+VertexXYZ convertToXYZ(const Vertex& v) {
+    return VertexXYZ{ v.xyz[0], v.xyz[1], v.xyz[2] };
+}
+
+std::vector<VertexXYZ> convert(std::vector<Vertex> v) {
+    std::vector<VertexXYZ> result(v.size());
+    std::transform(v.begin(), v.end(), result.begin(), convertToXYZ);
+    return result;
+}
+
+std::vector<Vertex> createRing(int nSegments, float radius, glm::vec4 colors) {
+    const int nVertices = nSegments + 1;
+    std::vector<Vertex> vertices(nVertices);
+
+    const float fsegments = static_cast<float>(nSegments);
+
+    for (int i = 0; i <= nSegments; ++i) {
+        const float fi = static_cast<float>(i);
+
+        const float theta = fi * glm::pi<float>() * 2.0f / fsegments;  // 0 -> 2*PI
```

Comment:
We wouldn't write the extra `0` here to make it more immediately obvious that it's a float, not a double